### PR TITLE
Fix EZP-23644: add proper handling for hidden nodes in subtree fetches a...

### PR DESCRIPTION
**Pull request by paulborgermans**
...nd multiple locations
Tested on 5.3 and 5.4

Link: https://jira.ez.no/browse/EZP-23644

## TODO
- [x] Rebase commits

### CODE REVIEW

**Comment by andrerom** [comment]
review ping @yannickroger @bdunogier

File: classes/ezfezpsolrquerybuilder.php [hunk 1]
```diff
@@ -180,6 +180,7 @@ public function buildSearch( $searchText, $params = array(), $searchTypes = arra
             $params['Filter'] = array( $params['Filter'] );
         }
 
+        $pathFieldName = $ignoreVisibility ? eZSolr::getMetaFieldName( 'path' ) : eZSolr::getMetaFieldName( 'visible_path' ) ;
```
Comment (bdunogier) [inline classes/ezfezpsolrquerybuilder.php]: Could be shortened a bit with `$pathFieldName = eZSolr::getMetaFieldName( $ignoreVisibility ? 'path' : 'visible_path );`

**Comment by bdunogier** [comment]
+1

**Comment by yannickroger** [comment]
+1
I updated the description adding a reminder to squash commits.

Who's up to merge this ?

**Pull request merged**